### DISCUSSION TITLE
How can I add custom titles to untitled slides in the Reveal.js menu?

### DISCUSSION THREAD

**Discussion by danielvartan** [quarto-dev/quarto-cli#10879] [Q&A] [answered]
### Description

Hi there, 👋

I'm trying to set titles for slides that currently appear as untitled in the Reveal.js menu.

![image](https://github.com/user-attachments/assets/f8e6ecba-b984-4bf1-90d9-c39330461080)

Using Quarto, I can create untitled slides like the ones below, but the issue is that these untitled slides clutter the presentation menu.

````qmd
---
title: "Example"
format: revealjs
---

# Example

---

Some content

---

## {background-iframe="https://quarto.org/" background-interactive=true}

## Example

Some content
````

I’m aware of the `hideMissingTitles` option, which can hide these slides from the menu, but that’s not what I’m looking for. I need these slides to remain findable, just with appropriate titles

```yaml
format:
  revealjs:
    menu:
      hideMissingTitles: true
``` 

Does anyone have a solution for adding custom titles to untitled slides in the menu?

**Comment by mcanouil** [comment]
See related discussion and the "menu" plugin for Reveal.js documentation:

- https://github.com/quarto-dev/quarto-cli/discussions/10478

**Reply by danielvartan** [reply]
Thank you @mcanouil .

**Comment by danielvartan** [comment, marked as answer]
To assist others, since one of the goals of this discussion tab on GitHub is to build a database of answers, here’s the solution:

Simply add the `data-menu-title` attribute to the slide (see the example below):

````qmd
---
title: "Example"
format: revealjs
---

## {data-menu-title="Custom Menu Title 1"}

Some content.

## {data-menu-title="Custom Menu Title 2" background-iframe="https://quarto.org/" background-interactive=true}

---

Note: This will not work for slides created using `---`. 

In these cases, it's best to use an empty heading, like so: `# {data-menu-title="Custom Menu Title 3"}`.

---
````
